### ISSUE TITLE
Model creation exception results in failure with a status code error

### ISSUE THREAD

**Issue by marius-serban** [theappbusiness/TABResourceLoader#53]
In `NetworkResponseHandler.swift`:
```swift
switch HTTPURLResponse.statusCode {
    case 200..<400:
      let parsedResult = Result<Resource.Model> {
        return try resource.model(from: data)
      }
      switch parsedResult {
      case .success(let model):
        return .success(model, HTTPURLResponse)
      case .failure:
        return .failure(.statusCodeError(statusCode: HTTPURLResponse.statusCode), HTTPURLResponse)
      }
    default:
```
Instead, another error should be returned, something about parsing/decoding failing

**Comment by deanbrindley87** [comment]
@marius-serban i've just come across this problem too and agree it needs fixing. 
the problem is the `failure` case isn't capturing its `error` property and so errors that occur during the `resource.model(from: data)` method become lost.

i'll create a PR for it, thanks for raising!

**Comment by samdods** [comment]
@marius-serban is this not available in the `.statusCodeCustomError` case of `enum NetworkServiceError`?

**Comment by marius-serban** [comment]
@samdods in case of parsing failure, a `.statusCode` error is returned, surely this can't be right there's nothing wrong with the status code (it's 200). The `statusCodeCustomError` _could_ be used but, it's not expressing the actual error which is related to parsing.

**Comment by KaneCheshire** [comment]
Agree with @marius-serban, it's

**Comment by KaneCheshire** [comment]
Clicked the wrong button when commenting... sorry. :| 

What I was trying to say before foolishly clicking the wrong button, was that I agree with @marius-serban. We should leverage the expressive behaviour of `catch` in Swift which makes it clear what error occurred.

**Comment by samdods** [comment]
i don't disagree with this, but for context, i just want to highlight the [PR that removed the `.couldNotParseData` case](https://github.com/theappbusiness/TABResourceLoader/pull/51/files).

**Comment by KaneCheshire** [comment]
Ah I forgot about that.

**Comment by samdods** [comment]
i'm going to fix this now

**Comment by samdods** [comment]
i don't know what the rationale was for removing the `.couldNotParseData` case. but my proposed solution is that we add this back in.

**Comment by samdods** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [TABResourceLoader: Model creation exception results in failure with a status code error](https://trello.com/c/BHcRosWb/7-tabresourceloader-model-creation-exception-results-in-failure-with-a-status-code-error)

**Comment by KaneCheshire** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Issue #53: Model creation exception results in failure with a status code error](https://trello.com/c/lo1PMifh/13-issue-53-model-creation-exception-results-in-failure-with-a-status-code-error)

**Comment by samdods** [comment]
Fixed by PR #60

Available on CocoaPods in version 7.0.0
